The job **`dist-aarch64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/6368444102/job/17287805165) [(plain)](https://github.com/rust-lang-ci/rust/commit/2685d3e6283fc95374b65eb4dd1fadac622ab00e/checks/17287805165/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[TIMING] doc::RustbookSrc<bootstrap::doc::UnstableBook> { target: aarch64-pc-windows-msvc, name: "unstable-book", src: "C:\\a\\rust\\rust\\build\\aarch64-pc-windows-msvc\\md-doc\\unstable-book", parent: Some(UnstableBook { target: aarch64-pc-windows-msvc }) } -- 4.422
[TIMING] doc::UnstableBook { target: aarch64-pc-windows-msvc } -- 0.000
[TIMING] compile::StartupObjects { compiler: Compiler { stage: 0, host: x86_64-pc-windows-msvc }, target: x86_64-pc-windows-msvc } -- 0.000
thread 'main' panicked at lib.rs:1716:25:
could not read dir "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0\\bin\\rustlib\\x86_64-pc-windows-msvc\\bin": Os { code: 3, kind: NotFound, message: "The system cannot find the path specified." }
Build completed unsuccessfully in 0:01:14
  local time: Sun Oct  1 05:54:57 CUT 2023
  network time: Sun, 01 Oct 2023 05:54:57 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/src/bootstrap/compile.rs
+++ b/src/bootstrap/compile.rs
@@ -158,6 +158,23 @@ impl Step for Std {
         target_deps.extend(copy_third_party_objects(builder, &compiler, target));
         target_deps.extend(copy_self_contained_objects(builder, &compiler, target));
 
+        // The LLD wrappers and `rust-lld` are self-contained linking components that can be
+        // necessary to link the stdlib on some targets. We'll also need to copy these binaries to
+        // the `stage0-sysroot` to ensure the linker is found when bootstrapping on such a target.
+        if compiler.stage == 0 && compiler.host == builder.config.build {
+            // We want to copy the host `bin` folder within the `rustlib` folder in the sysroot.
+            let src_sysroot_bin = builder
+                .rustc_snapshot_sysroot()
+                .join("lib")
+                .join("rustlib")
+                .join(&compiler.host.triple)
+                .join("bin");
+            let target_sysroot_bin =
+                builder.sysroot_libdir(compiler, target).parent().unwrap().join("bin");
+            t!(fs::create_dir_all(&target_sysroot_bin));
+            builder.cp_r(&src_sysroot_bin, &target_sysroot_bin);
+        }
+
         let mut cargo = builder.cargo(compiler, Mode::Std, SourceType::InTree, target, "build");
         std_cargo(builder, target, compiler.stage, &mut cargo);
         for krate in &*self.crates {

